hammer/sim/xcelium/__init__.py

```diff
@@ -458,4 +556,337 @@ def sim_xrun(self) -> bool:
+      models = [os.path.normpath(modelfile.path) for modelfile in os.scandir(model_path)]
+
+      # Get normalized, absolute paths for analog schematic files
+      schematic_path = os.path.join(os.getcwd(), schematics_dir)
+      schematics = [os.path.normpath(schematic.path) for schematic in os.scandir(schematic_path)]
+
+      # Warnings for missing files.
+      if (len(schematics) > 0 and len(models) == 0):
+        logging.warning(f"No models found in {model_path} to support analog schematics.")
+      else:
+        if (len(models) == 0):
+          logging.warning(f"No models found in {model_path}.")
+        if (len(schematic_path) == 0):
+          logging.warning(f"No analog schematics found {schematics}.")
+
+      # Get string representation of AMS control file template.
+      template = t.read()
+
+      # Format model_paths list as a single string with include statements.
+      formatted_models = ""
+      for modelpath in models:
+        formatted_models += f"include {modelpath!r}\n"
+      # Format schematic_paths list as a single string with include statements.
+      formatted_schematics = ""
+      for schematicpath in schematics:
+        formatted_schematics += f"include {schematicpath!r}\n"
+
+      # Replace empty model deck with formatted string of model paths.
+      model_template = re.sub("// model deck\n", "// model deck\n" + formatted_models, template)
+
+      # Replace empty schematic deck with formatted string of schematic paths.
+      schematic_template = re.sub("// schematic deck\n", "// schematic deck\n" + formatted_schematics, model_template)
+
+      # Write filled template to AMS control file.
+      f.write(schematic_template)
+
+      # Close files properly.
+      t.close()
+      f.close()
+      return True
+  
+  def option_extractor(self, argfile_names=[]):
+    if not argfile_names:
+      return {}
+
+    opts = {}
+    
+    # Extract the options from each argfile listed, ignoring duplicate opts and file inclusions
+    for filename in argfile_names:
+      path = os.path.join(self.run_dir, filename) 
+      file_opts = {}
+      f = open(path, "r")
+
+      for line in f:
+        if (line[0] == "-"):
+          split_line = line.split(sep=None, maxsplit=2)
+          if (len(split_line) > 1):
+            opt_key, opt_arg = split_line[0].strip("- "), split_line[1].lstrip("\n")
+          else:
+            opt_key, opt_arg = split_line[0].strip("- "), ""
+          
+          file_opts[opt_key] = opt_arg
+
+      opts.update(file_opts)
+      f.close()
+
+    return opts
+
+  def scriptwriter(self, options, additional_options, collect=False, sourcedir="", blacklist=[], sourcelist=[]):
+    """
+    Writes all prepared files and arguments to the run_mxh shell script
+    """
+    runpath = os.path.join(self.run_dir, "run_mxh")
+
+    f = open(runpath, "w+")
+
+    # Write Shebang + xrun clean
+    f.write("#!/bin/csh -f\n#\nxrun -clean \\\n")
```

Comment:
Does this only work in csh? Generally we want users to use bash family terminals.